internal/services/containerapps/helpers/container_apps.go

```diff
@@ -2214,6 +2216,20 @@ func SecretsSchema() *pluginsdk.Schema {
 		Sensitive: true,
 		Elem: &pluginsdk.Resource{
 			Schema: map[string]*pluginsdk.Schema{
+				"identity": {
+					Type:        pluginsdk.TypeString,
+					Optional:    true,
+					Sensitive:   false,
+					Description: "The identity to use for accessing key vault reference.",
+				},
+
+				"key_vault_url": {
+					Type:        pluginsdk.TypeString,
+					Optional:    true,
+					Sensitive:   true,
+					Description: "The key vault reference URL for this secret.",
+				},
```

Comment:
Can we instead define it as below, by:

1. Renaming it to `key_vault_secret_id` to keep consistency across the code base (as you can find `key_vault_key_id` used similarly)
2. Adding the validation function
3. No need to mark it as sensitive

```suggestion
				"key_vault_secret_id": {
					Type:        pluginsdk.TypeString,
					Optional:    true,
					ValidateFunc: keyVaultValidate.NestedItemIdWithOptionalVersion,
					Description: "The id of the key vault secret.",
				},
```

Reply:
Great, I've updated those three items, and have updated the property name throughout (docs pending)